scripts/Muon/GUI/MuonAnalysis/loadfile/load_file_view.py

```diff
@@ -0,0 +1,129 @@
+        self.file_path_edit.setToolTip("")
+        self.file_path_edit.setObjectName("filePathEdit")
+
+        self.horizontal_layout = QtGui.QHBoxLayout(self)
+        self.horizontal_layout.setObjectName("horizontalLayout")
+        self.horizontal_layout.addWidget(self.file_path_edit)
+        self.horizontal_layout.addWidget(self.browse_button)
+
+        self.setLayout(self.horizontal_layout)
+
+    def on_browse_clicked(self, slot):
+        self.browse_button.clicked.connect(slot)
+
+    def on_file_edit_changed(self, slot):
+        self.file_path_edit.returnPressed.connect(slot)
+
+    def show_file_browser_and_return_selection(self, file_filter, search_directories, multiple_files=False):
+        default_directory = search_directories[0]
+        if multiple_files:
+            chosen_files = QtGui.QFileDialog.getOpenFileNames(self, "Select files", default_directory,
+                                                              file_filter)
+            return [str(chosen_file) for chosen_file in chosen_files]
+        else:
+            chosen_file = QtGui.QFileDialog.getOpenFileName(self, "Select file", default_directory,
+                                                            file_filter)
+            return [str(chosen_file)]
+
+    def disable_loading(self):
+        self.disable_load_buttons()
+        self.loadingStarted.emit()
+
+    def enable_loading(self):
+        self.enable_load_buttons()
+        self.loadingFinished.emit()
+        self.dataChanged.emit()
+
+    def notify_loading_started(self):
+        self.loadingStarted.emit()
+
+    def notify_loading_finished(self):
+        self.loadingFinished.emit()
+        self.dataChanged.emit()
+
+    def disable_load_buttons(self):
+        self.browse_button.setEnabled(False)
+        self.file_path_edit.setEnabled(False)
+
+    def enable_load_buttons(self):
+        self.browse_button.setEnabled(True)
+        self.file_path_edit.setEnabled(True)
+
+    def get_file_edit_text(self):
+        if self._store_edit_text:
+            return str(self._stored_edit_text)
+        else:
+            return str(self.file_path_edit.text())
+
+    def set_file_edit(self, text, store=False):
+        if store:
+            self._store_edit_text = True
+            self._stored_edit_text = text
+            self.file_path_edit.setText("(... more than 10 files, use right-click -> copy)")
+        else:
+            self.file_path_edit.setText(text)
+        self._cached_text = self.get_file_edit_text()
+
+    def clear(self):
+        self.set_file_edit("")
+        self._store_edit_text = False
+        self._cached_text = ""
+
+    def reset_edit_to_cached_value(self):
+        tmp = self._cached_text
+        self.set_file_edit(tmp)
+        self._cached_text = tmp
+
+    def warning_popup(self, message):
+        # TODO : limit the number of warnings to prevent spamming if a list of bad files is selected.
+        QtGui.QMessageBox.warning(self, "Error", str(message))
```

Comment:
you can use https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/message_box.py

Reply:
Yes, I will use that. As you can see from the TODO here, I was hoping to find a simple way to limit the number of boxes that can appear. I think warning messages cease to be useful if they start spamming the window.

I was unable to find a clean way to do it though. Any ideas?

Reply:
Could you have a flag for a failed load, and instead of calling the pop up you just add the file to a list. Then if the flag is true you pass the list of failed runs to the message box?